nodeserver/controllers/routes.js

```diff
@@ -0,0 +1,83 @@
+var fs = require('fs');
+var path = require('path');
+var http = require('http');
+var bodyParser = require('body-parser');
+
+module.exports = function(server){
+
+  server.use(bodyParser.urlencoded({ extended: true }));
+
+  /**
+  * Controller that routes requests based on the URL using ExpressJS.
+  * Server runs on PORT 3000.
+  **/
+
+  // The root of the project is index.html, which is opened when the project is opened.
+  server.get('/', function(request, response){
+    response.sendFile(path.resolve('./public/index.html'));
+  });
+
+  // This API call can be used to get the current data.
+  // Reads data from JSON file and returns it.
+  // Using a JSON file since the data is small. Can be extended to DB.
+  server.get('/api/get/publications', function(request, response){
+    var fileContent = fs.readFileSync('data/publications.json');
+    var jsonContent = JSON.parse(fileContent);
+    response.json(jsonContent);
+  });
+
+  // This API call can be used to post a new publication to the current data.
+  // Reads data from JSON file, appends new data and redirects to root.
+  server.post('/api/post/newpublication', function(request, response){
+      var newPublication = {
+        "id": parseInt(request.body.id),
+        "title": request.body.title,
+        "author": request.body.author,
+        "year": parseInt(request.body.year),
+        "citations": parseInt(request.body.citations)
```

Comment:
nit: Our client uses trailing commas to help maintain "blame" and consistency. It's a little weird at first but you get used to it. We should follow same conventions.